src/VAC/VectorAnimationComplex/VAC.cpp

```diff
@@ -1277,6 +1277,18 @@ void VAC::smartDelete_(const CellSet & cellsToDelete)
         smartDeleteCell(ivertex);
 }
 
+ShapeType VAC::shapeType(const CellSet & cells)
+{
+ // TODO - multi selection
+    KeyVertexSet keyVertices = cells;
+    for( auto vertex : keyVertices)
+    {
+        ShapeType type =  vertex->shapeType();
+        return type;
+    }
+    return ShapeType::NONE;
```

Comment:
I think it makes more sense to have the shape type on faces and not on vertices unless we need them on vertices as well. But I don't think we will need that for vertices right now.

Reply:
what is faces? which part of shape it represent?

Reply:
if it represent whole shape, then its good to have shape type in it

Reply:
A face is created with each shape, so we have them for circles, squares etc. The exception right now is the polygons which don't create a face. But I think that is something we will implement soon. So I think it makes more sense to use faces since some faces, like circles, will have tons of vertices and this might become inefficient with lots of geometry.

Reply:
You can see the faces as part of the shapes but they have much lower opacity. If you select a shape currently the opacity will increase on the face.

Reply:
Do we create face for all shapes? we create face only when if (global()->isDrawShapeFaceEnabled()). In what cases, it is disabled.

Reply:
I think would be better to store all of the drawn shapes as separate objects with: 
ShapeID, ShapeType, CurrentPrinthead(color), and list of all cells(vertex, edges and faces).
In this case we will have actual information for each shapes and will be abble to select related shape if user select it on shapes list of layer widget.

Reply:
> Do we create face for all shapes? we create face only when if (global()->isDrawShapeFaceEnabled()). In what cases, it is disabled.

We need it for infill so we should create faces for all shapes and I think we do as well.

Reply:
curve shape does not have face. Any reason?

Reply:
> I think would be better to store all of the drawn shapes as separate objects with: ShapeID, ShapeType, CurrentPrinthead(color), and list of all cells(vertex, edges and faces). In this case we will have actual information for each shapes and will be abble to select related shape if user select it on shapes list of layer widget.

It feels redundant to have this. Each face will already have information on edges and vertices. The only place it's needed is when we have polylines. I guess in that case the edge(s) are needed as well if we want to be able to show them in the sidebar with each layer.

Reply:
yes, agreed. no need have redundant info and manage it

Reply:
> curve shape does not have face. Any reason?

In the VPaint application (you can download on the vpaint github) you can create faces from circles as well as straight lines. So it should be possible.

Reply:
Currently, we are not supporting polyline. do we have plan? I tried by adding face in curve and its working. The only difference is it show very very light color face

Reply:
I think for the time being, we can have shapeType in faces if it solve the current issue and create a separate task on to how to keep shape type in different way.